### PR TITLE
egressgw: emit a warning rather than a fatal error when L7 proxy is enabled

### PR DESCRIPTION
**Pull request by jibi**
The egress gateway feature is only partially incompatible with
--enable-l7-proxy as traffic will not be forwarded to an egress gateway
only if there's an L7 policy selecting the same endpoint selected by an
egress gateway policy.

Because of this, just logging a warning should be sufficient as in most
cases both features can be enabled and used at the same time.

### CODE REVIEW

**Review by raphink — APPROVED** [review]

**Review by pchaigno — COMMENTED** [review]
I think we should also change [the deployment step in the getting started guide](https://docs.cilium.io/en/latest/gettingstarted/egress-gateway/#enable-egress-gateway) to have `--set l7Proxy=false`. This is just to avoid users running into a warning after following our guide. In general, we want to avoid normalizing warnings.

**Comment by jibi** [comment]
> I think we should also change [the deployment step in the getting started guide](https://docs.cilium.io/en/latest/gettingstarted/egress-gateway/#enable-egress-gateway) to have `--set l7Proxy=false`. This is just to avoid users running into a warning after following our guide. In general, we want to avoid normalizing warnings.

Yep, I'll add that with the rest of the docs changes

**Review by pchaigno — APPROVED** [review]

File: daemon/cmd/daemon.go [hunk 1]
```diff
@@ -899,8 +899,8 @@ func NewDaemon(ctx context.Context, cancel context.CancelFunc, epMgr *endpointma
 		}
 
 		if option.Config.EnableL7Proxy {
-			return nil, nil, fmt.Errorf("egress gateway requires L7 proxy to be disabled (--%s=\"false\").",
-				option.EnableL7Proxy)
+			log.Warning("both egress gateway and L7 proxy are enabled. This is currently not fully supported: " +
```
Comment (julianwiedmann) [inline daemon/cmd/daemon.go]: Maybe
```suggestion
			log.Warning("both %s and %s are enabled. This is currently not fully supported: " +
			"if the same endpoint is selected both by an egress gateway and a L7 policy, endpoint traffic will not go through egress gateway.", option.EnableIPv4EgressGateway, option.EnableL7Proxy)
```

**Review by julianwiedmann — APPROVED** [review]

File: daemon/cmd/daemon.go (hunk 1 above)
Comment (jibi) [reply daemon/cmd/daemon.go]: updated while trying to keep it consistent with the other messages :+1:

**Comment by joestringer** [comment]
Do we have a github issue to track this issue and solve the underlying feature incompatibility? It would be nice to also link that issue so that if users see this and realize it's a problem for them, they have a direct link to find and provide feedback. In the past we've added `logfields.URL:         "https://github.com/cilium/cilium/pull/...",` into log messages to give such hints to users.

**Comment by brb** [comment]
Set the 1.12 release blocker label to allow RC users to use L7 with EgressGW.

**Comment by jibi** [comment]
> Do we have a github issue to track this issue and solve the underlying feature incompatibility? It would be nice to also link that issue so that if users see this and realize it's a problem for them, they have a direct link to find and provide feedback. In the past we've added `logfields.URL: "https://github.com/cilium/cilium/pull/...",` into log messages to give such hints to users.

it's just documented in the docs for now, I'll create and link an issue :+1:

**Review by joestringer — APPROVED** [review]

**Pull request merged**